File: 剑指offer/easy/20210914/findContinuousSequence.cpp
class Solution {
public:
    vector<vector<int>> findContinuousSequence(int target) {
        vector<vector<int>> res;
        vector<int>ans;
        for(int l=1,r=2;l<r;){
            int sum=(l+r)*(r-l+1)/2;
            if(sum==target){
                ans.clear();
                for(int i=l;i<=r;i++){
                    ans.emplace_back(i);
                }
                res.emplace_back(ans);
                l++;
            }else if(sum<target){
                r++;
            }else{
                l++;
            }
        }
        return res;
    }
};

File: Review/20211004/findContinuousSequence.go
package _0211004

func findContinuousSequence(target int) [][]int {
	res:=[][]int{}
	left,right:=0,0
	for right<target{
		sum:=right*(right+1)/2-left*(left+1)/2
		if sum>target{
			left++
		}else if sum<target{
			right++
		}else{
			temp:=[]int{}
			for i:=left+1;i<=right;i++{
				temp=append(temp,i)
			}
			res=append(res,temp)
			left++
			right++
		}
	}
	return res
}
